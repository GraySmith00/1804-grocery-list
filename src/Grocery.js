import React from 'react'

const Grocery = ({name, quantity, starred, onPurchase}) => {
  return (
    <article className={`Grocery ${starred ? 'starred' : ''}`}>
      <h2>Item: {name}</h2>
      <h3>Quantity: {quantity}</h3>
      <button onClick={onPurchase} className='purchase-btn'>Purchase</button>
    </article>
  )
}

export default Grocery
