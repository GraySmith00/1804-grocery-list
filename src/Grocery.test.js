import React from 'react'
import { shallow, mount } from 'enzyme'
import Grocery from './Grocery'

describe('Grocery', () => {
  it('should have the text of banana in the h2', () => {
    // Setup
    const wrapper = shallow(<Grocery name='banana'/>)
    const expected = <h2>Item: banana</h2>

    // Execution
      
    // Expectation
    expect(wrapper.contains(expected)).toEqual(true)
  })

  it('should match the snapshot', () => {
    // Setup
    const wrapper = shallow(
      <Grocery 
        name='apple'
        quantity={6}
        starred={false}
      />
    )

    // Execution && Expectation
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has the class of starred if the starred prop is true', () => {
    // Setup
    const wrapper = shallow(
      <Grocery 
        name='apple'
        quantity={6}
        starred={true}
      />
    )
    
    // Execution && Expectation
    expect(wrapper.is('.starred')).toEqual(true)
  })

  it('hasn\'t the class of starred if the starred prop is false', () => {
    // Setup
    const wrapper = shallow(
      <Grocery 
        name='apple'
        quantity={6}
        starred={false}
      />
    )
    
    // Execution && Expectation
    expect(wrapper.is('.starred')).toEqual(false)
  })

  it('calls onPurchase when the purchase button is clicked', () => {
    // Setup
    const onPurchaseMock = jest.fn()
    const wrapper = shallow(
      <Grocery 
        name='apple'
        quantity={6}
        starred={false}
        onPurchase={onPurchaseMock}
      />
    )

    // Execution
    wrapper.find('.purchase-btn').simulate('click')

    // Expectation
    expect(onPurchaseMock).toHaveBeenCalled()
  })

})
