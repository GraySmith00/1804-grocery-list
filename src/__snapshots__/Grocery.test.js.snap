// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grocery should match the snapshot 1`] = `"<article class=\\"Grocery \\"><h2>Item: apple</h2><h3>Quantity: 6</h3><button class=\\"purchase-btn\\">Purchase</button></article>"`;
